src/Meshes/abstractMesh.ts

```diff
@@ -1305,11 +1306,19 @@ export class AbstractMesh extends TransformNode implements IDisposable, ICullabl
 
     /** @hidden */
     public _getPositionData(applySkeleton: boolean, applyMorph: boolean): Nullable<FloatArray> {
-        var data = this.getVerticesData(VertexBuffer.PositionKind);
+        let needUpdatePositionInSkeletonPass = false;
+        let data = this.getVerticesData(VertexBuffer.PositionKind);
 
         if (data && ((applySkeleton && this.skeleton) || (applyMorph && this.morphTargetManager))) {
             data = Tools.Slice(data);
             this._generatePointsArray();
+            needUpdatePositionInSkeletonPass = !!(applySkeleton && this.skeleton) && !(applyMorph && this.morphTargetManager);
```

Comment:
Why not updating the vertices based on the skeleton if there are also some morph targets? The morph may not apply on all vertices, so it's possible some vertices are still updated by the bones even where there is both a skeleton and some morph targets?

Reply:
yeah you are right I missed the part where morph could be on different targets. 
I wanted to save the copy but I believe we cannot